subsys/power/include/pm_policy.h

```diff
@@ -0,0 +1,182 @@
+/*
+ * Copyright (c) 2020 Intel corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef _PM_POLICY_H_
+#define _PM_POLICY_H_
+
+#include <string.h>
+#include <sys/util.h>
+#include <power_state.h>
+#include <toolchain/common.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief Power management policy
+ * @defgroup Pm_policy Power management policy
+ * @{
+ * @}
+ */
+struct pm_policy;
+
+/**
+ * @brief Power management policy API.
+ */
+struct pm_policy_api {
+	/* initialize the power management policy */
+	void (*init)(void);
+
+	/* get next system power state based on current system
+	 * status and the power management policy where this
+	 * api resides.
+	 */
+	pm_state_t (*next_state)(struct pm_policy *policy);
```

Comment:
Is there a way to communicate the time of the next known event to this, as currently done with `_sys_suspend()`?

Reply:
@pabigot if the policy need timeout value to make decision, just use timeout API directly.